head.go

```diff
@@ -520,15 +530,13 @@ func (h *Head) Init(minValidTime int64) error {
 
 		sr := wal.NewSegmentBufReader(s)
 		err = h.loadWAL(wal.NewReader(sr), multiRef)
-		sr.Close() // Close the reader so that if there was an error the repair can remove the corrupted file under Windows.
+		if err := sr.Close(); err != nil {
+			level.Warn(h.logger).Log("msg", "error while closing the wal segments reader", "err", err)
+		}
 		if err == nil {
 			continue
 		}
-		level.Warn(h.logger).Log("msg", "encountered WAL error, attempting repair", "err", err)
-		h.metrics.walCorruptionsTotal.Inc()
-		if err := h.wal.Repair(err); err != nil {
-			return errors.Wrap(err, "repair corrupted WAL")
-		}
+		return err
```

Comment:
If I am not wrong, if we were able to repair here before, we would continue to loading next segment. But with this, if we encounter corruption, we return will repair it in `DB.Open()`, but if that was a success, we miss the further segments as we don't load them.

Reply:
after a short discussion we figured that this is a bug in the current code and this PR accidentally fixes it.

All WAL files after the corrupted one should be ignored to avoid adding misleading data.